src/calendar/tests/CalendarMonth.spec.js

```diff
@@ -0,0 +1,232 @@
+import React from 'react/addons';
+import CalendarMonth from '../CalendarMonth';
+import CalendarDate from '../CalendarDate';
+import moment from 'moment';
+import {} from 'moment-range';
+import _ from 'underscore';
+
+
+const TestUtils = React.addons.TestUtils;
+
+describe('The CalendarMonth Component', function () {
+
+  beforeEach(function () {
+
+    const getCalendarMonth = (props) => {
+
+      props = _.extend({
+        firstOfWeek: 0,
+        firstOfMonth: this.firstOfMonth,
+        enabledRange: moment.range(moment(), moment().add(3, 'years')),
+        dateComponent: CalendarDate,
+        disableNavigation: false,
+        dateRangesForDate: function () {
+          return {
+            count: function () {
+              return props.count || 1;
+            },
+            getIn: function (data) {
+              if (data[0] === 0) {
+                return '#333';
+              }
+              return '#444';
+            },
+          };
+        },
+        onMonthChange: function () {},
+        onYearChange: function () {},
+      }, props);
+
+
+      return (<CalendarMonth {...props} />);
+    };
+
+    this.useShallowRenderer = (props) => {
+      this.shallowRenderer = TestUtils.createRenderer();
+      this.shallowRenderer.render(getCalendarMonth(props));
+      this.renderedComponent = this.shallowRenderer.getRenderOutput();
+      this.container = this.renderedComponent.props.children[0];
+      this.table = this.renderedComponent.props.children[1];
+    };
+
+    this.useDocumentRenderer = (props) => {
```

Comment:
If I understand correctly, `componentWillUnmount` won't get called through this setup mechanism. So this may be a potential problem in the future.

I think we should be good citizens and add an afterEach hook to use `React.unmounComponentAtNode`. Not a blocker to being merged.